### ISSUE TITLE
Add action is resulting in a "FAILED" log

### ISSUE THREAD

**Issue by tychota** [amwmedia/node-plop#94]
# The problem

```failure
Cannot read property 'toString' of undefined
```

![image](https://user-images.githubusercontent.com/13785185/48146769-0b852300-e2b6-11e8-9c8f-c7a7615cd3dc.png)

# The repro (versions and co)

My blog:

https://github.com/tychota/blog/tree/257dbb6b553a745c691e9eb4eb4602a0f191d55f

You can find the package.json and the plopfile

# The start of the investigation and the hack

```diff
patch-package
--- a/node_modules/node-plop/lib/generator-runner.js
+++ b/node_modules/node-plop/lib/generator-runner.js
@@ -111,7 +111,7 @@ exports.default = function (plopfileApi, flags) {
 		// show the resolved value in the console
 		result => ({
 			type: cfg.type || 'function',
-			path: _colors2.default.blue(result.toString())
+			path: _colors2.default.blue(result && result.toString())
 		}),
 		// a rejected promise is treated as a failure
 		function (err) {
```

used with patch-package fix the problem.

See https://github.com/tychota/blog/blob/257dbb6b553a745c691e9eb4eb4602a0f191d55f/patches/node-plop%2B0.16.0.patch

# The proper fix

I will be super please to PR a better fix if you have an idea. I must understand why in this case result is undefined. And then fix so result of Add action so the root cause is fixed.

Would be kind if you can help me going in the right direction.

If I don't answser in github, you can ping me in twitter (@tychota) ! Github notifications is quite messy for me.

**Comment by amwmedia** [comment]
Thanks for pointing this out. The issue you are seeing is actually originating from your Custom Action (Function) that is running prior to the `add` action. This functions are supposed to return a `String` or a `Promise` that resolves with a string. The string is the result that will be show in the final readout when plop completes.

So, the short term fix is to make that function return an empty string. The long term fix is for plop to handle `undefined` in a better way.

**Comment by tychota** [comment]
Hello, I haven't seen the notification earlier :)

Thank you for the hint !

**Comment by amwmedia** [comment]
fixed as of 0.17.x
